From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_git@polytechnique.org>
Date: Wed, 25 May 2016 20:53:05 +0200
Subject: [PATCH] {BUG} drm/i915: fix 32-bit shift overflow

clang reports:

    drivers/gpu/drm/i915/i915_gem_stolen.c:114:16: error: signed shift
    result (0xFFFF00000) requires 37 bits to represent, but 'int' only
    has 32 bits [-Werror,-Wshift-overflow]
                    base = bsm & BSM_MASK;
                                 ^~~~~~~~
    drivers/gpu/drm/i915/i915_reg.h:90:28: note: expanded from macro
    'BSM_MASK'
    #define   BSM_MASK (0xFFFF << 20)
                        ~~~~~~ ^  ~~
---
 drivers/gpu/drm/i915/i915_reg.h | 2 +-
 include/drm/i915_drm.h          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 5c06413ae0e6..d3500323c436 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -87,7 +87,7 @@ static inline bool i915_mmio_reg_valid(i915_reg_t reg)
 #define   DEVEN_MCHBAR_EN (1 << 28)
 
 #define BSM 0x5c
-#define   BSM_MASK (0xFFFF << 20)
+#define   BSM_MASK (0xFFF << 20)
 
 #define HPLLCC	0xc0 /* 85x only */
 #define   GC_CLOCK_CONTROL_MASK		(0x7 << 0)
diff --git a/include/drm/i915_drm.h b/include/drm/i915_drm.h
index b1755f8db36b..1be2a7995965 100644
--- a/include/drm/i915_drm.h
+++ b/include/drm/i915_drm.h
@@ -93,6 +93,6 @@ extern bool i915_gpu_turbo_disable(void);
 #define    I845_TSEG_SIZE_1M	(3 << 1)
 
 #define INTEL_BSM 0x5c
-#define   INTEL_BSM_MASK (0xFFFF << 20)
+#define   INTEL_BSM_MASK (0xFFF << 20)
 
 #endif				/* _I915_DRM_H_ */
-- 
